cloud/wfm/src/task/BaseTask.ts

```diff
@@ -0,0 +1,64 @@
+import { EventEmitter } from 'eventemitter3';
+import {Result} from '../result';
+import { Task, TaskEventData, TaskStatus } from './index';
+
+/**
+ * Base implementation for {@link Task}s
+ * Derived classes are expected to implement their run {@link #run} methods
+ */
+class BaseTask extends EventEmitter implements Task {
+  public result: Result;
+  /**
+   * Storage field for the `status` property
+   */
+  protected _status: TaskStatus | number = TaskStatus.PENDING;
```

Comment:
avoid `_` as property name

`TaskStatus | number` - so custom statuses will be just numbers our customers will map by themselves? 
That's really clever :P